tesseract_python/tesseract_python/swig/tesseract_collision/types.i

```diff
@@ -0,0 +1,201 @@
+  res = PyTuple_New(2);
+  if (!ConvertFromEigenToNumPyMatrix<Eigen::Vector3d >(&v1, &$1[0]))
+  {
+    Py_XDECREF(res);
+    SWIG_fail;
+  }
+  PyTuple_SetItem(res, 0, v1);
+  if (!ConvertFromEigenToNumPyMatrix<Eigen::Vector3d >(&v2, &$1[1]))
+  {
+    Py_XDECREF(res);
+    SWIG_fail;  
+  }
+  
+  PyTuple_SetItem(res, 1, v2);
+  $result = res;
+}
+
+%typemap(in, fragment="Eigen_Fragments") Eigen::Vector3d [2]  (std::array<Eigen::Vector3d,2> temp, PyObject* v1, PyObject* v2)
+{
+  // Convert from tuple to Eigen::Vector3d[2]
+  if (!PyTuple_Check($input))
+    SWIG_fail;
+  if (PyTuple_Size($input) != 2)
+    SWIG_fail;
+  v1 = PyTuple_GetItem($input, 0);
+  v2 = PyTuple_GetItem($input, 1);
+
+  if (!ConvertFromNumpyToEigenMatrix<Eigen::Vector3d>(&temp[0], v1))
+    SWIG_fail;
+  if (!ConvertFromNumpyToEigenMatrix<Eigen::Vector3d>(&temp[1], v2))
+    SWIG_fail;  
+  $1 = &temp[0];
+}
+
+namespace tesseract_collision
+{
+using CollisionShapesConst = std::vector<tesseract_geometry::Geometry::ConstPtr>;
+using CollisionShapeConstPtr = tesseract_geometry::Geometry::ConstPtr;
+using CollisionShapePtr = tesseract_geometry::Geometry::Ptr;
+
+enum class ContinouseCollisionType
+{
+  CCType_None,
+  CCType_Time0,
+  CCType_Time1,
+  CCType_Between
+};
+
+enum class ContactTestType
+{
+  FIRST = 0,
+  CLOSEST = 1,
+  ALL = 2,
+  LIMITED = 3
+};
+
+
+
+
+struct ContactResult
+{
+  double distance;
+  int type_id[2];
+  std::string link_names[2];
+  int shape_id[2];
+  int subshape_id[2];
+  Eigen::Vector3d nearest_points[2];
+  Eigen::Vector3d normal;
+  Eigen::Vector3d cc_nearest_points[2];
+  double cc_time;
+  ContinouseCollisionType cc_type;
+
+  ContactResult();
+  void clear();
+};
+
+// TODO: These using statements are not working
+//using ContactResultVector = tesseract_common::AlignedVector<ContactResult>;
+//using ContactResultMap = tesseract_common::AlignedMap<std::pair<std::string, std::string>, ContactResultVector>;
+using ContactResultVector = std::vector< tesseract_collision::ContactResult,Eigen::aligned_allocator< tesseract_collision::ContactResult > >;
+using ContactResultMap = std::map< std::pair< std::string,std::string >,std::vector< tesseract_collision::ContactResult,Eigen::aligned_allocator< tesseract_collision::ContactResult > >,std::less< std::pair< std::string,std::string > >,Eigen::aligned_allocator< std::pair< std::pair< std::string,std::string > const,std::vector< tesseract_collision::ContactResult,Eigen::aligned_allocator< tesseract_collision::ContactResult > > > > >;
+}
+
+%inline
+{
+//std::size_t flattenResults(ContactResultMap&& m, ContactResultVector& v);
+//TODO: this function has lousy performance
+tesseract_collision::ContactResultVector flattenResults(tesseract_collision::ContactResultMap m)
+{
+    tesseract_collision::ContactResultVector v;
+    tesseract_collision::flattenResults(std::move(m),v);
+    return v;
+}
+}
+
+namespace tesseract_collision
+{
+/*struct ContactTestData
```

Comment:
May not want to expose this. This should only be used internally by the collision libraries.

Reply:
We need access to the collision checker from Python. The contact monitor example required access to these classes. Is there a specific type you don't want exposed?

Reply:
Just the ContactTestData. Everything else should be good.